Migrate to karma from mocha

clbond opened the pull request.

clbond commented:
@SethDavenport

SethDavenport commented on package.json:
```diff
@@ -20,19 +19,8 @@
-    "branches": 80,
-    "functions": 80
+    "test": "karma start --singleRun true",
+    "test:watch": "karma start --singleRun false"
```
So there's no longer a coverage threshold? That's OK - we can add it back when we address https://github.com/rangle/rangle-starter/issues/8

clbond replied:
Yessir, that's correct. I will add it back, it's very easy: istanbul check-coverage --statement N --branch N --function N

SethDavenport commented on package.json:
```diff
@@ -108,7 +107,6 @@
     "require-hacker": "^2.1.2",
     "rimraf": "^2.4.3",
     "sinon": "^1.17.3",
```
sinon is dedundant with jasmine

SethDavenport replied:
chai too

clbond replied:
Alright, I will remove sinon and chai today

SethDavenport commented on src/components/form/index.test.js:
```diff
@@ -26,14 +25,19 @@ describe('Form', () => {
-    assert.isTrue(onBlur.calledOnce, 'focused element was not blurred');
-  });
+  // NOTE(cbond): This test is wild unsafe ... it attempts to monkey-patch
+  // window.document(!) and then does not restore it afterward, raising the
```
yes - can you log an issue for this in rangle-starter?

SethDavenport commented on webpack.config.js:
```diff
@@ -22,27 +27,36 @@ module.exports = {
 
-  devtool: 'source-map',
+  devtool: process.env.NODE_ENV === 'production' ?
+    'source-map' :
```
`!devmode` instead?

SethDavenport commented:
As long as it works on IE9 (subject to known CSS issue https://github.com/rangle/rangle-starter/issues/64), :shipit:

clbond commented:
@SethDavenport Alright, removed sinon and chai dependencies and converted all assertions / spies to Jasmine syntax, and reintroduced coverage thresholds

clbond commented:
@SethDavenport Good call - my changes broke the project on IE9. (Actually it caused a complete IE crash so it was difficult to debug what the issue was). But changing back to es5-shim from babel-polyfill appears to fix the issue. It runs successfully on IE9 now.
